lib/buffer.js

```diff
@@ -115,6 +115,9 @@ function assertSize(size) {
     err = new TypeError('"size" argument must be a number');
   else if (size < 0)
     err = new RangeError('"size" argument must not be negative');
+  else if (size > binding.kMaxLength)
+    err = new RangeError('"size" argument must not be larger ' +
+                         'than 0x' + binding.kMaxLength.toString(16));
```

Comment:
Just trying to be close to the old behavior here(https://github.com/nodejs/node/blob/9cd44bb2b683446531306bbcff8739fc3526d02c/lib/buffer.js#L174). Not sure why it did that either. Personally I prefer a base-10 number too. Will change it that way.